lib/global_id/locator.rb

```diff
@@ -106,7 +106,9 @@ def use(app, locator = nil, &locator_block)
 
       private
         def locator_for(gid)
-          @locators.fetch(normalize_app(gid.app)) { default_locator }
+          @locators.fetch(normalize_app(gid.app)) do
+            gid.respond_to?(:unscoped) ? ACTIVE_RECORD_LOCATOR : DEFAULT_LOCATOR
```

Comment:
Whoops, you're right - `gid.model_class` it is.

Though I don't agree with using `defined?(ActiveRecord::Base) && gid < ActiveRecord::Base`, I find duck typing to be a better win.